src/sentry/tasks/summaries/daily_summary.py

```diff
@@ -0,0 +1,187 @@
+from sentry.services.hybrid_cloud.user_option import user_option_service
+from sentry.silo import SiloMode
+from sentry.snuba.referrer import Referrer
+from sentry.tasks.base import instrumented_task, retry
+from sentry.tasks.summaries.utils import (
+    ONE_DAY,
+    DailySummaryProjectContext,
+    OrganizationReportContext,
+    fetch_key_error_groups,
+    fetch_key_performance_issue_groups,
+    project_event_counts_for_organization,
+    project_key_errors,
+    project_key_performance_issues,
+    user_project_ownership,
+)
+from sentry.types.activity import ActivityType
+from sentry.types.group import GroupSubStatus
+from sentry.utils.dates import floor_to_utc_day, to_datetime, to_timestamp
+from sentry.utils.outcomes import Outcome
+from sentry.utils.query import RangeQuerySetWrapper
+
+
+# The entry point. This task is scheduled to run every day at 4pm PST.
+@instrumented_task(
+    name="sentry.tasks.daily_summary.schedule_organizations",
+    queue="reports.prepare",
+    max_retries=5,
+    acks_late=True,
+    silo_mode=SiloMode.REGION,
+)
+@retry
+def schedule_organizations(timestamp: float | None = None, duration: int | None = None) -> None:
+    if timestamp is None:
+        # The time that the report was generated
+        timestamp = to_timestamp(floor_to_utc_day(timezone.now()))
+
+    if duration is None:
+        # The total timespan that the task covers
+        duration = ONE_DAY
+
+    organizations = Organization.objects.filter(status=OrganizationStatus.ACTIVE)
+    for organization in RangeQuerySetWrapper(
+        organizations, step=10000, result_value_getter=lambda item: item.id
+    ):
+        if features.has("organizations:daily-summary", organization):
+            user_ids = {
+                user_id
+                for user_id in OrganizationMember.objects.filter(
+                    organization_id=organization.id, teams__projectteam__project__isnull=False
+                ).values_list("user_id", flat=True)
+            }
+
+            # TODO: convert timezones to UTC offsets and group
+            users_by_tz = defaultdict(list)
+            users_with_tz = user_option_service.get_many(
+                filter=dict(user_ids=user_ids, key="timezone")
+            )
+            # if a user has not set a timezone, default to UTC
+            users_without_tz = set(user_ids) - {uo.user_id for uo in users_with_tz}
+            if users_with_tz:
+                users_by_tz["UTC"] = list(users_without_tz)
+            for uo in users_with_tz:
+                users_by_tz[uo.value].append(uo.user_id)
+            for tz in users_by_tz.keys():
+                # Create a celery task per timezone
+                prepare_summary_data(timestamp, duration, organization.id)
+
+
+@instrumented_task(
+    name="sentry.tasks.daily_summary.prepare_summary_data",
+    queue="reports.prepare",
+    max_retries=5,
+    acks_late=True,
+    silo_mode=SiloMode.REGION,
+)
+@retry
+def prepare_summary_data(
```

Comment:
Or i guess - maybe just need a better understanding who the target audience is here

Reply:
I think one thing that might help for milestone 2 is to make a `Summary` data structure - so we could call different functions and gather the data in that - to simplify this function.

That'd be able to move things like top projects and things like that into the `Summary` model and make it a little simpler to me.

Reply:
It will be sent to an individual user, `user_project_ownership` checks which projects the user has access to and when we actually build the message, we can use that to filter the results.